app/src/main/kotlin/com/wire/android/ui/home/messagecomposer/MessageComposer.kt

```diff
@@ -218,182 +228,294 @@ private fun MessageComposer(
-                        }
-                        .wrapContentSize()) {
-                    Divider()
-                    Box(Modifier.wrapContentSize()) {
-                        transition.AnimatedVisibility(
-                            visible = { messageComposerState.messageComposeInputState != MessageComposeInputState.Enabled },
-                            // we are animating the exit, so that the MessageComposeActions go down
-                            exit = slideOutVertically(
-                                targetOffsetY = { fullHeight -> fullHeight / 2 }
-                            ) + fadeOut()
-                        ) {
-                            MessageComposeActions(messageComposerState, focusManager)
-                        }
-                    }
+                if (!isUserBlocked) {
+                    // Box wrapping the SendActions so that we do not include it in the animationContentSize
+                    // changed which is applied only for
+                    // MessageComposerInput and CollapsingButton
+                    SendActions(
+                        Modifier.Companion.constrainAs(sendActions) {
+                            bottom.linkTo(additionalActions.top)
+                            end.linkTo(parent.end)
+                        },
+                        messageComposerState,
+                        transition,
+                        onSendButtonClicked
+                    )
+                    // Box wrapping MessageComposeActions() so that we can constrain it to the bottom of MessageComposerInput and after that
+                    // constrain our SendActions to it
+                    MessageComposeActionsBox(
+                        Modifier.Companion
+                            .constrainAs(additionalActions) {
+                                top.linkTo(messageInput.bottom)
+                                bottom.linkTo(parent.bottom)
+                            },
+                        transition,
+                        messageComposerState,
+                        focusManager
+                    )
                 }
             }
             // Box wrapping for additional options content
             // we want to offset the AttachmentOptionsComponent equal to where
             // the device keyboard is displayed, so that when the keyboard is closed,
             // we get the effect of overlapping it
-            if (messageComposerState.attachmentOptionsDisplayed) {
-                Box(
+            if (messageComposerState.attachmentOptionsDisplayed && !isUserBlocked) {
+                AttachmentOptions(
+                    keyboardHeightOffSet,
+                    messageComposerState,
+                    onSendAttachment,
+                    onMessageComposerError,
+                    isFileSharingEnabled,
+                    tempCachePath
+                )
+            }
+        }
+    }
+}
+
+@Composable
+private fun AttachmentOptions(
```

Comment:
Maybe this is a good moment to rename and extract this to a separate `MessageAttachmentOptions` file?